### DISCUSSION TITLE
Improvements to ListRecords at scale

### DISCUSSION THREAD

**Discussion by JohnRoux** [filamentphp/filament#13466] [Ideas]
There are a couple of things in Filament which are amazing, but don't really work well when you have a lot of data in a table.

Eg: If you have 10m+ records, doing the count(*) to get the total count of records can get reeeeeally slow, and it's not super intuitive as to how to fix that.

Additionally, searching with indexing is hard. But if you swap to `$search%` you can still get a benefit of at least 1 more level of index lookup.

This can make a pretty huge impact.

Currently I've got a `CustomListRecords` page which I use, it has simple options like:
```
    protected bool $simplePagination = false;
    protected bool $splitSearchTerms = true;
```

So all you need to do is change those flags on your ListRecords page, and you've swapped out.

Is this something that I should make a PR for? Super happy to if that's the case.

Either way, I'd love to contribute a bit towards some Filament at Scale documentation :+1:

**Comment by danharrin** [comment]
An API to swap to simple pagination has been implemented in v4. I am converting this into an issue to track `splitSearchTerms()` for v4 also

**Reply by JohnRoux** [reply]
Champion!
Do we have a rough date in the pipeline for that btw?

**Reply by danharrin** [reply]
It will be about a month after Tailwind v4 is released
